### DISCUSSION TITLE
PreProcessing Issue

### DISCUSSION THREAD

**Discussion by standenman** [medspacy/medspacy#240] [General]
I seek to parse/extract data from a pdf file that is a collection of medical records from a particular source and for a particular person.  It strikes me that the data in these medical records MUST be tied to the treatment date.  When a doctor says "patient reports with a raging fever" we have to know when the doctor made this observation.  But I see very little in my research about this issue - the issue of "harvesting" the date that a given medical record pertains to.  There are just a ton of complications: sometimes we have "Visit Date", others times "Encounter Date" and other times the visit date is simply the big large font date at the top of the page.  And of course with medical records, we have all kinds of dates that could get in the way - dates that are not the treatment date.  And then sometimes we have a medical record for 1/1/23 that includes an observation for 1/1/22 or something.  The issue of what date a particular chunk of data applies to seems very complicated,  Again maybe I am missing something, but I just see nothing about this issue.  Am I barking up the wrong tree?

**Comment by burgersmoke** [comment]
You are right that identifying the date of any given assertion (whether a sign, symptom, lab test, etc) within a clinical document is a hard problem.  In fact, in the documents that I work with, it can often be the case that there may be an assertion (e.g., a COVID diagnosis) and there is no mention of a specific date or even a relative or broad range of time.

So yes, this is a hard problem.

In practice, we often find it best as much as possible to use structured data that should exist in the EHR for dates and times.  Even if these might not match when the actual event in the real world was made, trying to associate clinical events with dates from text only is a very hard problem.

So you have clinical documents only?  You don't have metadata about when the visit took place?  Or even when the note was entered?

You can attempt to extract and associate events with dates, but it is a hard problem.  And sometimes like I said, you will not even have the  text data there that even a human could identify.

There are many academic papers on this topic and many previous data challenges (i2b2, n2c2, etc) where research groups have attempted to solve this.  It can never be perfect, but in some instances it can be good.

Here is just one of many, many papers on the topic and you will see that indeed this is not a trivial problem:
https://www.sciencedirect.com/science/article/pii/S1532046413001299

**Reply by standenman** [reply]
Thank you so much for your reply.  Unfortunately, I do not have access the EHR - I just have the pdf file.

I am content with some assumptions though.  For example, if an assertion is made for a given treatment date I am going to assume that this assertion is as of the date of the treatment note, even though we know as a practical matter many statements seem to "bleed through" from prior notes, right?  Perhaps the template for the days note includes the completed field from the last note.  But anyway, if a clinician lets a statement make it through for a given date, I am going to assume it is "good" as of that treatment date. '

In other words, if it appears in medical notes for ##/##/#### it is a finding as of that date UNLESS it is a historical recounting.  For example, where the treatment note for 1/1/2023 includes something like: "1/1/2022: came into ER with a broken leg" then I don't want my app to give me the wrong data that on 1/1/2023 the patient came in to the ER with a broken leg.

I did take a look at the article you gave me - thank you for that.  I agree - very complicated.  I don't think I need that level - thank goodness - of detail.  Just have to figure out how do I discern for any given page what treatment date it belongs to, and then a strategy to "weed out" historical regurgitations.

**Comment by burgersmoke** [comment]
Yes, it is very common for findings from previous visits to be mentioned in clinical documentation.  In fact, it can often be copy-pasted from prior documents.  Side note: I have done a lot of research on the topic of copy-paste when I worked at 3M.  There's a lot on the topic out there: 
https://psnet.ahrq.gov/perspective/ehr-copy-and-paste-and-patient-safety

This is definitely a challenging problem to try to associate correct dates from text only.  Glad some of that literature was helpful.  If you follow other papers cited by that one or other papers which cite that paper you will see that it can be challenging to associate various kinds of assertions in clinical documents. Otherwise stated, medication dates might be written one way, vitals might be a different way, past medical history another way, etc.  It sometimes looks like different language dialects.

**Reply by standenman** [reply]
Thanks again for your help.  I will post here if I find a solution - that is, it will be kinda of a "down and dirty" solution as opposed perhaps to what a more clinical setting might require.  Your research looks quite interesting as well!
